src/qtt/data.py

```diff
@@ -1029,183 +1031,326 @@ def saveExperimentData(outputdir, dataset, tag, dstr):
     write_data(path, dataset)
 
 
-def makeDataSet1Dplain(xname, x, yname, y=None, xunit=None, yunit=None, location=None, loc_record=None):
-    ''' Make DataSet with one 1D array and one setpoint array
+def determine_parameter_unit(parameter):
+    """ Determine unit associated with a qcodes.parameter, otherwise None """
+    if isinstance(parameter, qcodes.Parameter):
+        return parameter.unit
+    else:
+        return None
+
+
+def _check_parameter(p):
+    """ Check p to be of correct type - used in MakeDataSet1D and MakeDataSet2D.
```

Comment:
Is a rename of the argument `p` to something like 'parameter' possible?

Reply:
Absolutely!